### ISSUE TITLE
Scribe-iOS Outreach

### ISSUE THREAD

**Issue by andrewtavis** [scribe-org/Scribe-iOS#181]
<a><img src="https://raw.githubusercontent.com/scribe-org/Organization/main/logo/ScribeAppLogo.png" width="25%" alt="Scribe Logo" align="right"></a>

## Sharing and promoting Scribe

This pinned issue is for discussing where and how best to promote Scribe-iOS. The purpose is to spread the word so that others can find Scribe as potential users, to provide feedback and to grow the contributors base. The issue text will be edited as the discussion below progresses :)

Everyone is welcome to share their inputs! Communities revolving around open-source, open-data, Wikidata, Wikimedia, Swift, iOS, iPadOS, language learning and all the languages Scribe supports should all be considered.

### Where Scribe is currently posted:

- [Wikimedia New Developers page](https://www.mediawiki.org/wiki/New_Developers)
- [Wikidata tools for lexicographical data](https://www.wikidata.org/wiki/Wikidata:Tools/Lexicographical_data)
- [open-source-ios-apps](https://github.com/dkhamsing/open-source-ios-apps)
- [up-for-grabs.net](https://github.com/up-for-grabs/up-for-grabs.net) [(pending)](https://github.com/up-for-grabs/up-for-grabs.net/pull/3271)

### Organizations that have featured Scribe:

- [Wikimedia](https://www.wikimedia.org/)
- [Wikimedia Deutschland](https://www.wikimedia.de/)
- [CocoaHeads Berlin](https://www.meetup.com/de-DE/cocoaheads-berlin/)

### General

- [Twitter](https://twitter.com/scribe_org) - suggested retweets and posts welcome!
- [Hacker News](https://news.ycombinator.com/)
- [Product Hunt](https://www.producthunt.com/)

### Tech Subreddits

|                                                        |                                                  |                                            |
| :----------------------------------------------------- | :----------------------------------------------- | :----------------------------------------- |
| [r/programming](https://www.reddit.com/r/programming/) | [r/software](https://www.reddit.com/r/software/)                                            | [r/apple](https://www.reddit.com/r/apple/)           |
| [r/ios](https://www.reddit.com/r/ios/)                 |                                                  [r/swift](https://www.reddit.com/r/swift/) | [r/opensource](https://www.reddit.com/r/opensource/) |

### Language Learning Subreddits

|                                                                  |                                                                |                                                      |
| :--------------------------------------------------------------- | :------------------------------------------------------------- | :--------------------------------------------------- |
| [r/languagelearning](https://www.reddit.com/r/languagelearning/) | [r/learnfrench](https://www.reddit.com/r/learnfrench/)         | [r/French](https://www.reddit.com/r/French/)         |
| [r/German](https://www.reddit.com/r/German/)                     | [r/italianlearning](https://www.reddit.com/r/italianlearning/) | [r/Portuguese](https://www.reddit.com/r/Portuguese/) |
| [r/russian](https://www.reddit.com/r/russian/)                   | [r/learnspanish](https://www.reddit.com/r/learnspanish/)       | [r/Spanish](https://www.reddit.com/r/Spanish/)       |
| [r/Svenska](https://www.reddit.com/r/Svenska/)                   |                                                                |                                                      |

**Comment by andrewtavis** [comment]
Something that could eventually be done is adding social media share buttons with custom messages in their links to the top of the [readme](https://github.com/scribe-org/Scribe-iOS/blob/main/README.md). This could be made into an issue if people think it would help 🤔

**Comment by andrewtavis** [comment]
I reached out to the folks at [cocoaheads.org](http://cocoaheads.org), and they said I'd be welcome to join the [Slack channel](http://cocoaheads.slack.com/) that they have. Will do that in the coming days :) Goals for that would be to network a bit to get the word out to other groups and their members, as well as see if there's interest in me doing more CocoaHeads presentations in the future (maybe remotely).

**Comment by andrewtavis** [comment]
Something we'll be considering in the future is [Outreachy](https://www.outreachy.org/), which is a program that would allow people to do internships while working on Scribe.

**Comment by andrewtavis** [comment]
Here are two more places on GitHub with repos for new developers:
- https://github.com/MunGell/awesome-for-beginners
- https://github.com/firstcontributions/first-contributions

**Comment by SaurabhJamadagni** [comment]
Let's keep hashtags to the minimum @andrewtavis. From what I have read, hashtags on twitter can sometimes negatively affect reach. Also I was wondering, what if we created a few Scribe intro tweets first before apple is done with the review so that people know what Scribe. Then the main v2.0.0 tweet could take place to make the real splash haha. Let me know what you think? Tweeting about v2.0.0 sounds like a great idea though! 🚀

**Comment by andrewtavis** [comment]
Suggestions for tweets are more than welcome, @SaurabhJamadagni! I can also send along the twitter PW to you as we discussed. Feel free to send along what it is that you're thinking for the various messages 😊

**Comment by andrewtavis** [comment]
> Something that could eventually be done is adding social media share buttons with custom messages in their links to the top of the [readme](https://github.com/scribe-org/Scribe-iOS/blob/main/README.md). This could be made into an issue if people think it would help 🤔

And let me know if you think social media buttons on the readme as discussed above would be useful. I was thinking that they could go in before the table of contents on the readme after the project description? 🤔 Also no need to add them if you don't see value in them though.

**Comment by SaurabhJamadagni** [comment]
> I can also send along the twitter PW to you as we discussed. Feel free to send along what it is that you're thinking for the various messages 😊

Yeah sure that works @andrewtavis! I'll try to craft a few tweets, maybe even a thread about everything that's new.

>  I was thinking that they could go in before the table of contents on the readme after the project description? 

Or I was also wondering, what about under the 'Download on the App Store' image? We could just have the socials logos that link to the respective pages.

**Comment by andrewtavis** [comment]
I was thinking under “Download on the App Store” would be too many buttons tight next to each other, but then that might be where they belong?

**Comment by SaurabhJamadagni** [comment]
> I was thinking under “Download on the App Store” would be too many buttons tight next to each other, but then that might be where they belong?

True @andrewtavis. Because I was thinking, once the reader starts reading the description and the rest of the texts, it would be a bit abrupt to put a Twitter button in between. We can just try out both positions actually. That should give us a better idea.

**Comment by andrewtavis** [comment]
Sounds good, @SaurabhJamadagni! I’m actually warming to the idea of putting it by the download button anyway :) Anything else besides Twitter and Reddit as of now? And do you use Reddit so we can check that the example text works?

**Comment by andrewtavis** [comment]
@SaurabhJamadagni, FYI on the Twitter front, [Cocoaheads Berlin](https://twitter.com/cocoaheads_BER) will be sending out something soon as one of the co-organizers was nice enough to offer 😊 Will be Hacktoberfest themed as he's also a participant :)

**Comment by SaurabhJamadagni** [comment]
> Anything else besides Twitter and Reddit as of now? And do you use Reddit so we can check that the example text works?

Do we need LinkedIn @andrewtavis? Maybe we could find some students or academics over there. Not sure, what do you think? Also, yes I do use reddit.

**Comment by SaurabhJamadagni** [comment]
> FYI on the Twitter front, [Cocoaheads Berlin](https://twitter.com/cocoaheads_BER) (the Meetup I go to) will be sending out something soon as one of the co-organizers was nice enough to offer

That sounds awesome! Glad to hear that 😊

**Comment by andrewtavis** [comment]
> Do we need LinkedIn @andrewtavis? Maybe we could find some students or academics over there. Not sure, what do you think?

I'd limit it to discussion style platforms as of now :) Maybe Discord, but I have no clue how that one really works. I just know it's popular with coders. 

> Also, yes I do use reddit.

Great! Just wanted to make sure we can check that the text is added. Let's think on what to include in the prompts then 🤔

**Comment by SaurabhJamadagni** [comment]
> I'd limit it to discussion style platforms as of now

Oh right right! Sorry @andrewtavis! Discord would be good too. But I think we could have it later. Based on the discussions we have on Twitter and Reddit, it would give us a better idea on what channels to have in the Discord server.

**Comment by andrewtavis** [comment]
No need to apologize, @SaurabhJamadagni! :) Sounds great about Discord 😊 Thanks for your insights!

I’ll look how to do custom text for a Reddit button and post it here :) Can I do the PR and you can review so you can check that the buttons work?

**Comment by SaurabhJamadagni** [comment]
> I’ll look how to do custom text for a Reddit button and post it here :) Can I do the PR and you can review so you can check that the buttons work?

That sounds good @andrewtavis 😊

**Comment by andrewtavis** [comment]
[Tweet](https://twitter.com/cocoaheads_ber/status/1583536546304888832?s=21) just went out from Cocoaheads Berlin, btw :)

**Comment by andrewtavis** [comment]
And a note on this front @SaurabhJamadagni, someone who was in an accelerator I was taking part in has a contact at the [Goethe-Institut](https://en.wikipedia.org/wiki/Goethe-Institut), so I'll hopefully be getting in contact with them soonish :) I think that the data issue should be solved before we do any major push into learning/culture institutions, but then we should have a solution for that in the next few weeks 😊

**Comment by SaurabhJamadagni** [comment]
> I think that the https://github.com/scribe-org/Scribe-iOS/issues/96 should be solved before we do any major push into learning/culture institutions, but then we should have a solution for that in the next few weeks 😊

Yes absolutely. That sounds great @andrewtavis! I have an exam coming this weekend, so I may be a bit inactive the coming half of this week (not as if I have been very active currently 🤪. Sorry, I am a bit swamped with a few different things). Thought I'll inform you in case I am unable to communicate later. But I'll definitely join you in bringing the SQLite changes in. I am very curious about it.

**Comment by andrewtavis** [comment]
Also trying to get Scribe featured on Tech blogs would be good :) The following could be considered:

- [ ] [TechCrunch](https://techcrunch.com/?guccounter=1)

**Comment by SaurabhJamadagni** [comment]
> Also trying to get Scribe featured on Tech blogs would be good :) The following could be considered:

Yes absolutely @andrewtavis! We could also try to write a Medium or a developer blog (Ex: dev.to) article highlighting the features of Scribe. Maybe even I'll learn some new features of Scribe along the way. Let me know what you think.

**Comment by andrewtavis** [comment]
> We could also try to write a Medium or a developer blog (Ex: dev.to) article highlighting the features of Scribe. Maybe even I'll learn some new features of Scribe along the way. Let me know what you think.

Sounds really great! Do you know what the process would be to get published there? :)

**Comment by SaurabhJamadagni** [comment]
> Sounds really great! Do you know what the process would be to get published there? :)

Yeah, for these platforms I think it is as easy as just creating a new post and posting it. I don't think that should be a challenge.

**Comment by andrewtavis** [comment]
Perfect :) Maybe post fixing the data issues, or that plus menu and then we can start reaching out?

**Comment by andrewtavis** [comment]
Now that we’re getting close to being done with the data issues 😊 Another thing to consider posting to is the [project part of the official Swift forums (link is to guidelines)](https://forums.swift.org/t/introducing-your-project/12861) :)

**Comment by andrewtavis** [comment]
The following is a list of communities I made that we could eventually reach out to 😊

General blogposts about communities:
- https://preply.com/en/blog/7-language-communities-to-improve-your-language-skills/
- https://blog.cometdocs.com/five-best-online-language-learning-communities

- [ ] https://podcasts.apple.com/gb/podcast/the-fluent-show/id780485096?mt=2
- [ ] [info@innovativelanguage.com](mailto:info@innovativelanguage.com) - 101 and survival phrases
- [ ] [info@linguistica360.com](mailto:info@linguistica360.com)
- [ ] https://www.dw.com/de/kontakt/s-30583
- [ ] https://coffeebreaklanguages.com/contactus/
- [ ] https://www.spotlight-verlag.de/kontakt
- [ ] https://www.easygerman.org/contact
- [ ] https://main.actualfluency.com/contact?r_done=1
- [ ] https://helenzaltzman.wordpress.com/contact/
- [ ] https://slate.com/podcasts/lexicon-valley
- [ ] https://slate.com/podcasts/spectacular-vernacular
- [ ] https://storylearning.com/contact
- [ ] https://survivalphrases.com/
- [ ] https://www.tandem.net/blog/top-free-podcasts-language-learners

**Comment by andrewtavis** [comment]
Ping @mhmohona, this is the issue I was talking to you about where we talk about outreach for Scribe :) Do you think one of these for Scribe-Data would also help?

**Comment by KesharwaniArpita** [comment]
Hi @andrewtavis , I had a simple idea for the outreach of Scribe community. I saw in the discussion that blogging is also an open field for outreach purpose.

> > We could also try to write a Medium or a developer blog (Ex: dev.to) article highlighting the features of Scribe. Maybe even I'll learn some new features of Scribe along the way. Let me know what you think.
> 
Building on this, what do you think about publishing a blog post series focused on Scribe? The series could explore the various communities within Scribe, highlight its features and data, share its vision and mission, reflect on achievements so far, and outline its future scope. I believe this would offer both visibility and an engaging story for readers.

Would love to hear your thoughts!

**Comment by andrewtavis** [comment]
Is certainly an interesting idea, @KesharwaniArpita! This is something that @mhmohona also had interest in as well :) I'm not sure how much capacity I'd have for writing, but I'd be happy to help review as needed. Where would you think to post the blogposts?

**Comment by andrewtavis** [comment]
By all means, @KesharwaniArpita! Thanks for your efforts here :)

**Comment by KesharwaniArpita** [comment]
Hi @andrewtavis , @mhmohona  I have just completed with the [blog on Scribe](https://github.com/KesharwaniArpita/BLOGS/blob/main/Scribe%20Data.md). Can you tell me your thoughts on it and what can be improved here? Post that we can simply publish to on medium and/or DEV community?

**Comment by andrewtavis** [comment]
Quick notes on it, @KesharwaniArpita :)

- First and foremost, really well written! 👏
- Could you not mention me specifically by name? Yes I did create Scribe, but at this point it wouldn't be nearly what it is without a lot of people putting in tons of effort, so I'd like to not be singled out. In a similar way the creators of Scribe are always listed as "The Scribe community" in Wikidata events :)
- Can we make the quote from me just a normal sentence given the above and credit it to the community?
- The plan for Scribe-Desktop is that it's written in Rust, so it should be `Kotlin and Rust` when talking about it and Scribe-Android 😊
- Similarly let's say "The Scribe community" rather than "Andrew and the Scribe team"

@mhmohona, @Jag-Marcel: we need to get your two's blogposts onto the readme sections in the projects btw. Would be great if you two could maybe reply to here with a link to all of your GSoC blogposts? Plan is that we'll expand the readmes for now and then we can add all of these to the website at some point 😊

Thanks so much, @KesharwaniArpita!

**Comment by mhmohona** [comment]
Hey @KesharwaniArpita, I really enjoyed reading your blog. And the blog repo itself is a good way to show all skill in writing. ✨ 

@andrewtavis, here are my blog links - 

- https://medium.com/@mhmohona/first-month-as-a-gsoc-intern-building-scribe-data-cli-d0c12c9e8371
- https://medium.com/@mhmohona/halfway-there-my-gsoc-adventure-with-scribe-data-cli-2ffe6d727ecb
- https://medium.com/@mhmohona/the-final-stretch-gsoc-journey-with-scribe-data-1740084c958d
- https://medium.com/@mhmohona/ins-and-outs-of-scribe-data-cli-bd51202aa7c6

**Comment by KesharwaniArpita** [comment]
Made the changes in the blog :) Read [here](https://github.com/KesharwaniArpita/BLOGS/blob/main/Scribe%20Data.md) 😃

**Comment by andrewtavis** [comment]
Only thing I saw from my comments was that it should be `The Scribe community is`, not `are` :) Thanks for the quick changes, @KesharwaniArpita! Let us know what the link is when it's up :)

**Comment by Jag-Marcel** [comment]
@andrewtavis here are my blog posts:
- https://jagmarcel.hashnode.dev/my-first-experiences-with-gsoc
- https://jagmarcel.hashnode.dev/gsoc-2024-midterm-report
- https://jagmarcel.hashnode.dev/gsoc-2024-final-report
I still want to add images to them, I'll try to do that this week.

**Comment by KesharwaniArpita** [comment]
Hi @andrewtavis @mhmohona @Jag-Marcel and @SaurabhJamadagni [Here's](https://medium.com/@arpita151103/scribe-an-open-source-solution-for-language-learning-and-data-accessibility-092dab026fd6) the link to my blog about scribe!!! I hope you'll like reading it. Any suggestions are most welcome!

**Comment by KesharwaniArpita** [comment]
Yeah sure !! I did that for reference too. I'll just remove them. Give me few minutes...

**Comment by KesharwaniArpita** [comment]
Done. Thanks @andrewtavis

**Comment by andrewtavis** [comment]
Thank you, @KesharwaniArpita!

**Comment by andrewtavis** [comment]
Ok all, all readmes for projects and the org have had the blogposts added to them 😊 Thanks for all the efforts in writing them!

**Comment by kadirabdullah496-code** [comment]
> <img alt="Scribe Logo" width="30%" src="https://raw.githubusercontent.com/scribe-org/Organization/main/logo/ScribeLogoRounded.png">
> 
> ## Sharing and promoting Scribe-iOS
> This pinned issue is for discussing where and how best to promote Scribe-iOS. The purpose is to spread the word so that others can find Scribe-iOS as potential users, to provide feedback and to grow the contributors base. The issue text will be edited as the discussion below progresses :)
> 
> Everyone is welcome to share their input, and efforts to share Scribe-iOS are much appreciated! Communities revolving around open-source, open-data, Wikidata, Wikimedia, Swift, iOS, iPadOS, language learning and the languages Scribe-iOS supports should all be considered.
> 
> ### Where Scribe-iOS is currently posted:
> * [Wikimedia New Developers page](https://www.mediawiki.org/wiki/New_Developers)
> * [Wikidata tools for lexicographical data](https://www.wikidata.org/wiki/Wikidata:Tools/Lexicographical_data)
> * [open-source-ios-apps](https://github.com/dkhamsing/open-source-ios-apps)
> * [up-for-grabs.net](https://up-for-grabs.net/) ([GitHub](https://github.com/up-for-grabs/up-for-grabs.net))
> * [firstcontributions.github.io](https://firstcontributions.github.io/) ([GitHub](https://github.com/firstcontributions/first-contributions))
> * [24 Pull Requests](https://24pullrequests.com/projects) ([GitHub](https://github.com/24pullrequests/24pullrequests))
> * [CodeTriage](https://www.codetriage.com/) ([GitHub](https://github.com/codetriage/CodeTriage))
> 
> ### Organizations that have featured Scribe-iOS:
> * [Wikimedia](https://www.wikimedia.org/): new developers page and hackathons
> * [Wikimedia Deutschland](https://www.wikimedia.de/): Wikidata events, blogposts and social media
> * [CocoaHeads Berlin](https://www.meetup.com/de-DE/cocoaheads-berlin/): meetups and social media
> 
> ## Places where Scribe-iOS outreach could occur:
> ### General
> * [Mastodon](https://wikis.world/@scribe) - suggested shares and posts welcome!
> * [Hacker News](https://news.ycombinator.com/)
> * [Product Hunt](https://www.producthunt.com/)
> 
> ### Conferences
> * [Swift Conf](https://swiftconf.com/)
> * [Slash Berlin](https://www.slash.berlin/)
> 
> ### Language and learning institutions
> * Forums for places where people learn programming as new developers are welcome
> * Major language institutions for a given language
>   
>   * For example the [Goethe-Institut](https://www.goethe.de/de/index.html), [Gesellschaft für deutsche Sprache](https://gfds.de/), [Académie française](https://www.academie-francaise.fr/), [Institut de France](https://www.institutdefrance.fr/), etc
>   * See [List of language regulators on Wiki](https://en.wikipedia.org/wiki/List_of_language_regulators)
>   * See [Higher education institutions that teach Swift](https://developer.apple.com/learn/institutions/)
> * University language learning departments for the languages that Scribe-iOS supports
> * Podcasts and other groups that support/serve language learners
> * Institutions that are trying to preserve languages as a Scribe-iOS keyboard could be made for them
> 
> ### Tech subreddits
> [r/programming](https://www.reddit.com/r/programming/)	[r/software](https://www.reddit.com/r/software/)	[r/opensource](https://www.reddit.com/r/opensource/)	[r/apple](https://www.reddit.com/r/apple/)
> [r/ios](https://www.reddit.com/r/ios/)	[r/swift](https://www.reddit.com/r/swift/)		
> ### Language subreddits
> [r/languagelearning](https://www.reddit.com/r/languagelearning/)	[r/learnfrench](https://www.reddit.com/r/learnfrench/)	[r/French](https://www.reddit.com/r/French/)	[r/German](https://www.reddit.com/r/German/)
> [r/italianlearning](https://www.reddit.com/r/italianlearning/)	[r/Portuguese](https://www.reddit.com/r/Portuguese/)	[r/russian](https://www.reddit.com/r/russian/)	[r/learnspanish](https://www.reddit.com/r/learnspanish/)
> [r/Spanish](https://www.reddit.com/r/Spanish/)	[r/Svenska](https://www.reddit.com/r/Svenska/)
